src/prefect/tasks/secrets/vault_secret.py

```diff
@@ -0,0 +1,161 @@
+        super().__init__(**kwargs)
+
+    def _get_vault_secret(self, name: str):
+        """
+        Connect to vault service and retrieve a secret value
+        Arguments:
+            - name (str): name of the secret to be retrieved using "<mount_point>/<path>"
+
+        Returns:
+            - secret (dict): a dict of the secret key/value items retrieved
+
+        Raises:
+            - ClientError: unable to configure the vault client
+            - PermissionError: unable to authenticate or attempting an unsupported auth method
+            - KeyError: unable to lookup secret in vault using the secret path
+            - PermissionError: provided token/role not authorised to access the secret
+        """
+        self.logger.debug(f"looking up vault path: {name}")
+        client = hvac.Client()
+        # get vault address url
+        vault_url = os.getenv("VAULT_ADDR", default=None)
+        if vault_url is None:
+            vault_url = os.getenv("vault_addr", default=None)
+            if vault_url is None:
+                raise ClientError(
+                    "VAULT_ADDR url var not found. "
+                    'Either "VAULT_ADDR" or "vault_addr" env var required.'
+                )
+        client.url = vault_url
+        self.logger.debug(f"vault addr set to: {client.url}")
+
+        # get vault auth credentials from the PrefectSecret
+        vault_creds = PrefectSecret(self.vault_credentials_secret).run()
+        if "VAULT_TOKEN" in vault_creds.keys():
+            client.token = vault_creds["VAULT_TOKEN"]
+        elif (
+            "VAULT_ROLE_ID" in vault_creds.keys()
+            and "VAULT_SECRET_ID" in vault_creds.keys()
+        ):
+            client.auth_approle(
+                vault_creds["VAULT_ROLE_ID"], vault_creds["VAULT_SECRET_ID"]
+            )
+        else:
+            raise ClientError(
+                "Unable to authenticate with vault service.  "
+                "Supported methods: token, appRole"
+            )
+        if not client.is_authenticated():
+            raise PermissionError(
+                "Unable to autheticate with vault using supplied credentials"
+            )
+        self.logger.debug("passed vault authentication check")
+        # regex to parse path into 2 named parts: <mount_point>/<path>
+        secret_path_re = r"^(?P<mount_point>[^/]+)/(?P<path>.+)$"
+        m = re.fullmatch(secret_path_re, name)
+        if m is None:
+            raise KeyError(
+                f'Invalid secret path: {name}.  Expected: "<mount_point>/<path>"'
+            )
+        vault_path = m.groupdict()
+        value = ""
+        try:
+            vault_secret = client.secrets.kv.v2.read_secret_version(
+                path=vault_path["path"], mount_point=vault_path["mount_point"]
+            )
+            value = vault_secret["data"]["data"]
+        except hvac.exceptions.InvalidPath as exc:
+            raise KeyError(f"Secret not found: {vault_path['path']}") from exc
```

Comment:
is it confusing to only see the path and not the mount_point/path here since the full thing is supplied to the task?